### PR TITLE
[M] CANDLEPIN-439: Added org-entity remapping to the refresh worker

### PR DESCRIPTION
**Pull request by Ceiu**
- Added checks to the refresh worker to determine whether or not
  an org's content or products are out of sync
- The refresh worker will now remap an org's entities if the org's
  mappings are determined to be out of sync
- Refactored the getProductsByOwner and getContentByOwner methods
  to no longer use CandlepinQuery. The original implementations have
  been temporarily maintained with a "CPQ" suffix, but are deprecated
  and should only be used by existing code; new code should use the
  non-CPQ variants instead

### CODE REVIEW

**Comment by Ceiu** [comment]
This should be ready to go, but I still want to check that we're not hitting an N+1 situation during the mapping step.

**Review by JoshAlbrecht23 — COMMENTED** [review]
+1 this looks good to me. A couple things that I thought was done well were:
- Good call on deprecated javadocs to the CPQ curator methods.
- Good logging for for when the remapping occurs which will make potential future troubleshooting easier.

File: src/main/java/org/candlepin/model/OwnerContentCurator.java [hunk 1]
```diff
@@ -481,6 +531,53 @@ public void removeOwnerContentReferences(Owner owner, Collection<String> content
...
+        log.debug("Removed {} owner-content mappings for owner: {}", rcount, owner);
+
+        if (contentIdMap != null) {
+            // TODO: content ID isn't part of the owner_content table, but it probably should be. Update
+            // this query to also include the content ID in the inserts
+            String sql = "INSERT INTO " + OwnerContent.DB_TABLE + " (owner_id, content_uuid) " +
+                "VALUES(:owner_id, :content_uuid)";
+
+            Query query = this.getEntityManager()
+                .createNativeQuery(sql)
```
Comment (Ceiu) [inline src/main/java/org/candlepin/model/OwnerContentCurator.java:568]: Note to self: add the native spaces query hint to this query.

**Review by Januson — APPROVED** [review]

File: src/main/java/org/candlepin/controller/refresher/mappers/AbstractEntityMapper.java [hunk 2]
```diff
@@ -102,72 +110,141 @@ public Set<String> getEntityIds() {
...
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean containsOnlyExistingEntityIds(Collection<String> ids) {
+        if (ids != null && !ids.isEmpty()) {
+            return !this.existingEntities.keySet()
+                .stream()
+                .anyMatch(id -> !ids.contains(id));
```
[the remark this answers is not in the document]
Comment (Ceiu) [reply src/main/java/org/candlepin/controller/refresher/mappers/AbstractEntityMapper.java:228]: Good catch. Fixed.

**Pull request merged**